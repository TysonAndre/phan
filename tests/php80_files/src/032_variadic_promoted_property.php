<?php
class VariadicPromotedProperty {
    public function __CONSTRUCT(public int $value, public ...$rest) {
    }

    public function other(public int $foo) {
    }
}
$c = fn(public int $x) => 123;
function fn32(public $var) {}
