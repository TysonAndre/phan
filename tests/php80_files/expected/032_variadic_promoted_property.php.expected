%s:3 PhanCompatibleConstructorPropertyPromotion Cannot use constructor property promotion before php 8.0 for public ...$rest of \VariadicPromotedProperty::__construct(int $value, ...$rest)
%s:3 PhanCompatibleConstructorPropertyPromotion Cannot use constructor property promotion before php 8.0 for public int $value of \VariadicPromotedProperty::__construct(int $value, ...$rest)
%s:3 PhanInvalidNode Cannot declare variadic promoted property
%s:6 PhanInvalidNode Cannot use visibility modifier on parameter public int $foo of non-constructor \VariadicPromotedProperty::other(int $foo)
%s:6 PhanPluginUseReturnValueNoopVoid The function/method \VariadicPromotedProperty::other() is declared to return (empty union type) and it has no side effects
%s:9 PhanCompatibleArrowFunction Cannot use arrow functions before php 7.4 in (fn)
%s:9 PhanInvalidNode Cannot use visibility modifier on parameter public int $x of non-constructor Closure(int $x)
%s:10 PhanInvalidNode Cannot use visibility modifier on parameter public $var of non-constructor \fn32($var)
%s:10 PhanPluginUseReturnValueNoopVoid The function/method \fn32() is declared to return void and it has no side effects
